xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/macros.vm

```diff
@@ -768,10 +768,7 @@ $fn
 #macro(panelheader $title)
 ## Note: We pass the Panel name as an HTML class attribute so that it's possible to style
 ## the Panels selectively using CSS.
-#set ($specialClassAttribute = '')
-#if ($paneldoc)
-  #set ($specialClassAttribute = $util.convertToAlphaNumeric($paneldoc.name))
-#end
+#set ($specialClassAttribute = $util.convertToAlphaNumeric($!paneldoc))
```

Comment:
* if paneldoc is not defined you'll get warnings in the console, why did you remove the "if"?
* you should use stringutils to convert to alpha and not the depredated old $util
* $paneldoc is a Document object. It's not a good idea to rely on the toString() method which can be changed at any time. You should get the documentReference and serialize it.
* you should keep backward compat too

Reply:
> if paneldoc is not defined you'll get warnings in the console, why did you remove the "if"?

Doesn't using `$!paneldoc` instead of `$!paneldoc` make it safe?
http://velocity.apache.org/engine/devel/user-guide.html#quiet-reference-notation

Reply:
> Doesn't using $!paneldoc instead of $paneldoc make it safe?

Only for printing AFAIK, i.e. when you use ``$!paneldoc`` inside a string. Here you're passing the value to a Java method so I don't think it applies, unless you do:

    $util.convertToAlphaNumeric("$!paneldoc"))

Reply:
@mflorea  You are right, null is passed to the method when no value was found. What you suggest should be just fine.

Reply:
Note: Regarding convertToAlphaNumeric if there's no simple api that exists in one of the velocity tool we provide, we could introduce our own StringTool that extends StringUtils for ex.

Reply:
Cool thanks @mflorea for catching the `$!paneldoc` issue.

Reply:
It's actually `org.xwiki.text.StringUtils`.

Reply:
Sure it's name it's not "StringTool" but it exactly follow your definition.

Reply:
Sure. I was just answering to your suggestion to introduce something that already exist :)